From 24547fb36377b3bbc39a91d887eb5161e73532e3 Mon Sep 17 00:00:00 2001
From: Christian Persch <chpe@src.gnome.org>
Date: Tue, 25 Apr 2023 23:06:40 +0200
Subject: [PATCH] widget: Don't consume right clicks on gtk4

gtk4 still needs to handle context menus correctly, but on stable branch
we need to remove always claiming the event since it doesn't do anything
yet.

https://gitlab.gnome.org/GNOME/vte/-/issues/2593
(cherry picked from commit ff1b03e450fddae623401d8dc619a6c4f17df42d)
---
 src/vte.cc | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/src/vte.cc b/src/vte.cc
index 76086d9d..b8e15d71 100644
--- a/src/vte.cc
+++ b/src/vte.cc
@@ -6829,16 +6829,6 @@ Terminal::widget_mouse_press(vte::platform::MouseEvent const& event)
 		break;
 	}
 
-#if VTE_GTK == 4
-        if (!handled &&
-            ((event.button() == vte::platform::MouseEvent::Button::eRIGHT) ||
-             !(event.modifiers() & (GDK_BUTTON1_MASK | GDK_BUTTON2_MASK)))) {
-                _vte_debug_print(VTE_DEBUG_EVENTS, "Showing context menu\n");
-                // FIXMEgtk4 context menu
-                handled = true;
-        }
-#endif /* VTE_GTK == 4 */
-
 	/* Save the pointer state for later use. */
         if (event.button_value() >= 1 && event.button_value() <= 3)
                 m_mouse_pressed_buttons |= (1 << (event.button_value() - 1));
-- 
GitLab

